### DISCUSSION TITLE
Can WIS and WAS run on the same machine?

### DISCUSSION THREAD

**Discussion by rarroyo6** [toverainc/willow-inference-server#141] [Q&A] [answered]
Can the WIS and WAS be run on the same machine?
And what would be the minimum or recommended hardware requirements for running the server (without counting the GPU, just the bare metal requirements)?

**Comment by nikito** [comment, marked as answer]
WAS and WIS can run on the same physical machine; since they are just containers you can have a docker server running on a machine that hosts both of them no problem. 🙂 In terms of minimum specs, WIS is generally GPU bound and isn't really affected by CPU as long as it's a decent one. Likewise WAS is it lightweight. For reference I am running WAS and WIS on an old i5 dual core hyperthreaded (so 3.4ghz with 4 total threads) and 16GB DDR4 RAM, and they run on a proxmox vm with GPU passthrough while I have another vm on the same hardware running home assistant. I set 2 cores to the docker vm and 2 cores to HA VM. 4GB RAM to HA, and 12GB to docker host. With all that the machine runs HA, WIS, and WAS perfectly fine with excellent performance.

**Reply by rarroyo6** [reply]
Great, thank you. :smiley:
I'm going to be setting up a WIS/WAS server in the next few days.
